src/lib/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { Fragment, useEffect, useMemo, useRef, useState } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import {
+  ModalWrapper,
+  ModalCard,
+  ModalBackdrop,
+  ModalHeader,
+  ModalTitle,
+  ModalContent,
+  ModalFooter,
+} from './style';
+import { Button } from '../Button';
+import idgen from '../../utils/idgen';
+import activeElement from '../../utils/activeElement';
+import ownerDocument from '../../utils/ownerDocument';
+import contains from '../../utils/contains';
+
+const ModalWithPortal = ({
+  open,
+  onClose,
+  children,
+  actions,
+  title,
+  disableBackdropClick,
+  disableEscapeKeyDown
+}) => {
+  const modalWrapperElement = useRef(null);
+  const modalCardElement = useRef(null);
+  const targetElement = useMemo(() => document.querySelector('body'));
+  const [triggerElement, setTriggerElement] = useState(null);
+
+  const onEscPress = ({ key }) => {
+    if (key === 'Escape') onClose()
+  }
+
+  const restoreLastFocus = () => {
+    if (triggerElement) {
+      triggerElement.focus();
+      setTriggerElement(null)
+    }
+  }
+
+  const autoFocus = () => {
+    const currentActiveElement = activeElement(ownerDocument(this));
+    if (modalWrapperElement && !contains(modalWrapperElement, currentActiveElement)) {
+      setTriggerElement(currentActiveElement)
+      modalCardElement.current.focus();
+    }
+  }
+
+  useEffect(() => {
+    document.body.style.overflow = open ? 'hidden' : 'unset';
+    document.body.onkeydown = (open && !disableEscapeKeyDown) ? onEscPress : null;
+
+    if (open) {
+      autoFocus()
+    }
+
+    if (!open) {
+      restoreLastFocus()
+    }
+  }, [open]);
+
+  const component = (
+    <ModalWrapper ref={modalWrapperElement} open={open}>
+      <ModalBackdrop
+        onClick={onClose}
+        disableBackdropClick={disableBackdropClick}
+        data-testid="modal-backdrop"
+      />
+      <ModalCard ref={modalCardElement} tabIndex="0" data-testid="modal-card">
+        {title && (
+          <ModalHeader>
+            <ModalTitle size="h6">{title}</ModalTitle>
+            <Button
+              icon="close"
+              secondary
+              onClick={onClose}
+              data-testid="modal-close-button"
+            />
+          </ModalHeader>
+        )}
+        <ModalContent>{children}</ModalContent>
+        {actions && (
+          <ModalFooter>{actions.map((action) => (
+            <Fragment key={`modal-action-${idgen()}`}>{action}</Fragment>
+          ))}
+          </ModalFooter>
+        )}
+      </ModalCard>
+    </ModalWrapper>
+  );
+
+  return ReactDOM.createPortal(component, targetElement);
```

Comment:
Acho que o componente deve ser renderizado apenas quando o estado do open for true, se não todas as modais dentro da página serão renderizadas sem necessidade e consumindo mais recurso do navegador sem necessidade afetando a performance, pois quem oculta a modal é o css.

Reply:
On it!